### ISSUE TITLE
A-C Support for ADM

### ISSUE THREAD

**Issue by athomasmoz** [mozilla-mobile/firefox-tv#2527]
### Why/User Benefit/User Problem

As an Amazon device user, I want to see tabs sent from my other devices on my TV right away, without having to ask for them on my TV

### Acceptance Criteria (how do I know when I’m done?)
* ADM is supported in AC (details via https://github.com/mozilla-mobile/android-components/issues/1918#issue-405527209)
  * Integrate SDK
  * Implement `concept-push`
  * Verify that ADM integration working
* [Graceful degradation](https://developer.amazon.com/docs/adm/integrate-your-app.html#gracefully-degrade-if-adm-is-unavailable) is supported in AC

jonalmeida is the a-c contact if you need assistance.

### Not included

* FFTV support, which is covered in separate ticket

**Comment by liuche** [comment]
From @psymoon:
Unfortunately ADM requires a `compileOnly` local `.jar` file as the integration (AFAIK there's no maven repo). Everything is working and built correctly; the unit tests are passing in AS.

However, when running `./gradlew :lib-push-amazon:test`, it can't seem to find ADM dependency.

```
java.lang.NoClassDefFoundError: com/amazon/device/messaging/ADMMessageHandlerBase
testCompileOnly: Additional dependencies only for compiling tests, not used at runtime
```

full log: https://taskcluster-artifacts.net/BSX6B50cSm-s2soE_HI5jA/0/public/logs/live_backing.log
new messages

tldr; integrating amazon stuff is a pain

**Comment by Baron-Severin** [comment]
> However, when running `./gradlew :lib-push-amazon:test`, it can't seem to find ADM dependency.

@psymoon Looks like `compileOnly` deps are never available to tests.  According to [this discussion on the Gradle board](https://discuss.gradle.org/t/compileonly-dependencies-are-not-available-in-tests/15366/3), the simple option is to redeclare the dependency using `testCompileOnly` (which is a pattern I think I've seen in projects before, even though it is a bit messy).

**Comment by psymoon** [comment]
https://github.com/mozilla-mobile/android-components/pull/3852 is now merged
